RandomObjects.py

```diff
@@ -91,17 +92,27 @@ def generateSeq(self):
         """Calculates a random size and location to generate a randomized shape
         then calls recursiveGen() many times to generate the shape
         """
-        sizeScalarW = int(math.sqrt(self.height) * 1.2)
+        sizeScalarW = int(math.sqrt(self.width) * 1.2)
         sizeScalarH = int(math.sqrt(self.height) * 1.2)
         sizeScalar = min(sizeScalarH, sizeScalarW)
         sizeScalar = random.randint(int(sizeScalar / 4), sizeScalar)
+        # robot's right most initial x position
+        midX_right = self.width/ 2 + robot_radius/ tile_size + 1
+        # robot's left most initial x position
+        midX_left = self.width / 2 - robot_radius/ tile_size - 1
+        # robot's down most initial y position
+        midY_down = self.height/ 2 - robot_radius/ tile_size - 1
+        # robot's up most initial y position
+        midY_up = self.height/ 2 + robot_radius/ tile_size + 1
         goodLoc = False
         while not goodLoc:
             randX = random.randint(0, self.width - sizeScalar)
             randY = random.randint(0, self.height - sizeScalar)
-            if (
-                    randY + sizeScalar >= self.height or randX + sizeScalar >= self.width or randY - sizeScalar <= 0 or randX - sizeScalar <= 0):
+            if randY + sizeScalar >= self.height or randX + sizeScalar >= self.width or randY - sizeScalar <= 0 or randX - sizeScalar <= 0:
                 goodLoc = False
+            # checks to see if the randX and randY is not overlapping with the robot's start position in the simulation
+            elif (midX_right >= randX >= midX_left) and (midY_up >= randY >= midY_down):
```

Comment:
I believe randX and randY are the CENTRAL coordinates of the new object being generated...? I don't think it is the EDGE of the object being generated. So for example lets say an object is generated where the center of the square Is NOT within this range (Not within the robot radius of the origin) BUT THE RIGHT MOST EDGE OF IT SITS ON THE ORGIIN! OHHHHHHH NOOOO MARIO! 

I think this will break right? Because an object can still have some part of it conflicting with the origin? Sorry I didn't think of this earlier... Am I correct? I could be wrong but I believe you will need to make a fix where you incorporate the dimensions of the generated objects